### PR TITLE
Replace CLI with same library docker CLI uses for maximum compatibility.

### PR DESCRIPTION
**Pull request by kekoav**
## What I did

I replaced the existing CLI interface with the same CLI library that is in use in docker/cli: [spf13/cobra](https://github.com/spf13/cobra) & [spf13/pflag](https://github.com/spf13/pflag).

### Changes

- [x] Replace `github.com/genuinetools/pkg/cli` with `github.com/spf13/cobra`. Combine flags where possible leveraging pflag enhancements.
- [x] Retain previous CLI behavior, and the general look-and-feel, including the version command.
- [x] Where applicable, move argument validation into separate function.
- [x] Update README with example help output.
- [x] Add support for a `img -v` or `img --version` to return short version like docker cli does. Long-form `img version` retains previous behavior.

## Why did I do that?

Img is touted as a drop-in replacement for `docker` commands. The easiest way for a user to adopt img is if the only modification to their code is `s/docker/img/g`. I'd like to improve compatibility and UX as much as possible to help drive adoption with no gotchas.

### Detailed reasons

1. CLI incompatibility generates a lot of issues, and is a headache for users wanting to adopt: #149, #203, #220, and maybe #201.
2. The Go flag module is notoriously unforgiving with argument order, and doesn't provide a clean POSIX interface. This is a problem [spf13/pflag](https://github.com/spf13/pflag) addresses, while not throwing out the way Go flag works. I tried to be as light-handed as possible with changing the way the app is structured.
3. By using the same library that docker/cli uses, we can more easily keep pace with CLI compatibility issues over the long term.
4. Cobra is an awesome library, it is well known and used by many major projects, so it's a solid choice on it's own merits.

### Other Feedback

I found it tricky to create a clean development environment to get the build working (all make commands). I added a Dockerfile.dev file which helped me along (based it off what I saw in travis config). Streamlining the process for a contributor would be helpful.

### CODE REVIEW

File: Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,76 @@
+FROM ubuntu:xenial
```
Comment (AkihiroSuda) [inline Dockerfile.dev]: why xenial?

File: Dockerfile.dev [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		pkg-config \
+		git \
+		curl \
+		ca-certificates \
+		libseccomp-dev \
+		parallel \
+		uidmap \
+	&& rm -rf /var/lib/apt/lists/*
+
+ENV GOLANG_VERSION 1.11.9
```
Comment (AkihiroSuda) [inline Dockerfile.dev]: why 1.11 ?

File: Dockerfile.dev [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		libc6-dev \
+		make \
+		wget \
+		pkg-config \
+		git \
+		curl \
+		ca-certificates \
+		libseccomp-dev \
+		parallel \
+		uidmap \
```
Comment (AkihiroSuda) [inline Dockerfile.dev]: Please install the latest revision from source

https://github.com/genuinetools/img/blob/master/Dockerfile#L20-L24

File: Dockerfile.dev (hunk 1 above)
Comment (kekoav) [reply Dockerfile.dev]: @AkihiroSuda saw it here: https://github.com/genuinetools/img/blob/master/.travis.yml#L2

File: Dockerfile.dev (hunk 2 above)
Comment (kekoav) [reply Dockerfile.dev]: Also, based on what is set in .travis.yml

File: Dockerfile.dev (hunk 3 above)
Comment (kekoav) [reply Dockerfile.dev]: @AkihiroSuda will do

Comment (kekoav) [reply Dockerfile.dev]: Updated to build the newuidmap/newgidmap similar to what is happening for alpine.

**Comment by jessfraz** [comment]
so sorry needs a rebase

**Comment by kekoav** [comment]
@jessfraz rebased, please check again.

**Comment by AkihiroSuda** [comment]
merged via #232

**Pull request closed without merging**